filter/prefix_tree.go

```diff
@@ -0,0 +1,203 @@
+package filter
```

Comment:
There are no file `prefix_tree_test.go` :) Please create it with all the necessary tests for this important data structure.

Reply:
Done